src/native/corehost/host_runtime_contract.h

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef __HOST_RUNTIME_CONTRACT_H__
+#define __HOST_RUNTIME_CONTRACT_H__
+
+#include <stddef.h>
+#include <stdint.h>
+
+#if defined(_WIN32)
+    #define HOST_CONTRACT_CALLTYPE __stdcall
+#else
+    #define HOST_CONTRACT_CALLTYPE
+#endif
+
+// Known host property names
+#define HOST_PROPERTY_RUNTIME_CONTRACT "HOST_RUNTIME_CONTRACT"
+#define HOST_PROPERTY_ENTRY_ASSEMBLY_NAME "ENTRY_ASSEMBLY_NAME"
+
+struct host_runtime_contract
```

Comment:
@vitek-karas @jkotas thoughts on this as the mechanism for https://github.com/dotnet/runtime/issues/77458?

Reply:
Do we need to version this? (The typical sizeof versioning I mean)
I know that technically hostpolicy should always be versioned with coreclr, so the contract should always match. But it's a cross module thing.